crates/router/src/connector/datatrans.rs

```diff
@@ -110,12 +112,21 @@ impl ConnectorCommon for Datatrans {
 
         event_builder.map(|i| i.set_response_body(&response));
         router_env::logger::info!(connector_response=?response);
-
+        let error_code = response
+            .error
+            .as_ref()
+            .map(|e| e.code.clone())
+            .unwrap_or(consts::NO_ERROR_CODE.to_string());
+        let error_message = response
+            .error
+            .as_ref()
+            .map(|e| e.message.clone())
+            .unwrap_or(consts::NO_ERROR_MESSAGE.to_string());
         Ok(ErrorResponse {
             status_code: res.status_code,
-            code: response.code,
-            message: response.message,
-            reason: response.reason,
+            code: error_code,
+            message: error_message.clone(),
+            reason: Some(error_message),
```

Comment:
Remove the `some` here as mentioned in the previous comment.

Reply:
reason is Option here
and 
error_message is String

Reply:
Ok